cmd/samples/common/sample_helper.go

```diff
@@ -72,11 +96,43 @@ func (h *SampleHelper) SetupServiceConfig() {
 
 	logger.Info("Logger created.")
 	h.Logger = logger
-	h.Scope = tally.NoopScope
+	h.ServiceMetricScope = tally.NoopScope
+	h.WorkerMetricScope = tally.NoopScope
+
+	if h.Config.Prometheus != nil{
+		reporter, err := h.Config.Prometheus.NewReporter(
+			prometheus.ConfigurationOptions{
+				Registry: prom.NewRegistry(),
+				OnError: func(err error) {
+					logger.Warn("error in prometheus reporter", zap.Error(err))
+				},
+			},
+			)
+		if err != nil {
+			panic(err)
+		}
+
+		h.WorkerMetricScope, _ = tally.NewRootScope(tally.ScopeOptions{
+			Prefix: "Worker_",
+			Tags:           map[string]string{},
+			CachedReporter: reporter,
+			Separator:      prometheus.DefaultSeparator,
+			SanitizeOptions: &sanitizeOptions,
+		}, 1*time.Second)
+
+		// NOTE: this must be a different scope with different prefix, otherwise the metric will conflict 
```

Comment:
I don't recall this to be an issue for M3. Is it only a prometheus thing?

Reply:
yeah very likely. The error is from deeper part in tally where it implements the interface for prometheus.